stdlib/pqueue.mli

```diff
@@ -0,0 +1,217 @@
+    preferred to repeated insertions with [add]).
+
+    It is fine to have several elements with the same priority.
+    It is guaranteed that the element returned by [min_elt] (or
+    [min_elt_opt]) is the one that is removed from the priority queue
+    by [pop_min] (or [pop_min_opt] or [remove_min]). This is important
+    in many algorithms, (e.g. when peeking at several priority queues
+    and then selecting one to remove from).
+
+    @since 5.3
+*)
+
+module type OrderedType =
+  sig
+    type t
+      (** The type of priorities. *)
+
+    val compare : t -> t -> int
+      (** A total ordering function to compare priorities.
+
+          This is a two-argument function [f] such that [f p1 p2] is
+          zero if the priorities [p1] and [p2] are equal, [f p1 p2] is
+          strictly negative if [p1] is smaller than [p2], and [f p1
+          p2] is strictly positive if [p1] is greater than [p2].
+
+          The generic structural comparison function {!Stdlib.compare}
+          is a suitable ordering function for priority types such as
+          [int] or [string]. *)
+  end
+(** Input signature of the functors {!MakeMinPriority} and
+    {!MakeMaxPriority}. *)
+
+module type Min =
+  sig
+
+    (** {1:pqueue Min-priority queues} *)
+
+    type 'a t
+    (** The type of priority queues. *)
+
+    type 'a elt
+    (** The type of priority queue elements. *)
+
+    val create: unit -> 'a t
+    (** Return a new priority queue, initially empty. *)
+
+    val length: 'a t -> int
+    (** Return the number of elements in a priority queue. *)
+
+    val is_empty: 'a t -> bool
+    (** [is_empty q] is [true] iff [q] is empty, that is, iff [length q = 0]. *)
+
+    val add: 'a t -> 'a elt -> unit
+    (** [add q x] adds the element [x] in the priority queue [q]. *)
+
+    val add_seq: 'a t -> 'a elt Seq.t -> unit
+    (** [add q s] adds the elements of [s] in the priority queue [q]. *)
+
+    exception Empty
+    (** Raised when {!min_elt}, {!pop_min} or {!remove_min} is applied
+        to an empty queue. *)
+
+    val min_elt: 'a t -> 'a elt
+    (** [min_elt q] returns an element in queue [q] with minimal priority,
+        without removing it from the queue,
+        or raises {!Empty} if the queue is empty. *)
+
+    val min_elt_opt: 'a t -> 'a elt option
+    (** [min_elt_opt q] returns an element in queue [q] with minimal priority,
+        without removing it from the queue, or returns [None] if the queue
+        is empty. *)
+
+    val pop_min: 'a t -> 'a elt
+    (** [pop_min q] removes and returns an element in queue [q] with
+        minimal priority, or raises {!Empty} if the queue is empty. *)
+
+    val pop_min_opt: 'a t -> 'a elt option
+    (** [pop_min_opt q] removes and returns an element in queue [q] with
+        minimal priority, or returns [None] if the queue is empty. *)
+
+    val remove_min: 'a t -> unit
+    (** [pop_min_opt q] removes an element in queue [q] with
```

Comment:
Typo: the comment mentions `pop_min_opt` instead of `remove_min`.

Reply:
Thanks! Fixed.